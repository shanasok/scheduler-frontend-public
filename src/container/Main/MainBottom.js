import React from 'react';

export class MainBottom extends React.Component {

    render() {
        return <div id="mainBottom">
            Main Bottom goes here
        </div>
    }
}