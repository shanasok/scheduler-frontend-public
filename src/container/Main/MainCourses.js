import React from 'react';
import {CourseOption} from "../CourseOption/CourseOption";

export function populateCourseOption(courses){
    if (courses && Array.isArray(courses)){
        return courses.map(course =>
            <CourseOption key={course.id} course={course}/>
        )
    }
}


export default class MainCourses extends React.Component {

    render() {
        const courseOptions = populateCourseOption(this.props.courses);

        return <div id="mainCourses">
            {courseOptions}
        </div>
    }
}