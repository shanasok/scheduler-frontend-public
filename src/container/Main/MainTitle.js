import React from 'react';

export class MainTitle extends React.Component {

    render() {
        return <div id="mainTitle">
            Click to see course content, materials and scheduling information.
        </div>
    }
}