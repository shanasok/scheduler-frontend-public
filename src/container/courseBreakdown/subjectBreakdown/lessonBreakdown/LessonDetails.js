import React from 'react';
import '../../../../css/lessonStyles.css';
import LessonViewSlideButton from "./LessonViewSlideButton";
import RibbonLessonName from "./RibbonLessonName";

class LessonDetails extends React.Component {

    render() {
        return <div className="lesson">
            {/*{this.props.lesson.name}*/}
            <RibbonLessonName name={this.props.lesson.name}/>
            <LessonViewSlideButton lesson={this.props.lesson}/>
        </div>
    }
}

export default LessonDetails;