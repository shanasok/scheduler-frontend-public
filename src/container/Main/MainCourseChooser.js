import React from 'react';
import { connect } from "react-redux";

import {MainBottom} from "./MainBottom";
import MainCourses from "./MainCourses";
import {MainTitle} from "./MainTitle";

import '../../css/CourseOptionStyles.css';

const MainCourseChooser = props => (
    <div id="mainCourseChooser">
        <MainTitle />
        <MainCourses courses={props.courses} />
        <MainBottom />
    </div>
);

const mapStateToProps = state => ({
    courses: state.courses,
});

export default connect(mapStateToProps)(MainCourseChooser)