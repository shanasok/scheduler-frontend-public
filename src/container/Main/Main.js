import React from 'react';
import {Switch, Route, BrowserRouter as Router} from "react-router-dom";
import MainCourseChooser from "./MainCourseChooser";
import CourseDetails from "../courseBreakdown/CourseDetails"
import LessonSlide from "../courseBreakdown/subjectBreakdown/lessonBreakdown/LessonSlide";
import NotFound from "../NotFound";

export class Main extends React.Component {

    render() {
        return <div id="main">
            <Router>
                <Switch>
                    <Route exact path="/" component={() => <MainCourseChooser />} />
                    <Route exact path="/course" component={() => <CourseDetails />} />
                    <Route exact path="/lesson" component={() => <LessonSlide />} />
                    <Route component={() => <NotFound/>} />
                </Switch>
            </Router>
        </div>
    }
}