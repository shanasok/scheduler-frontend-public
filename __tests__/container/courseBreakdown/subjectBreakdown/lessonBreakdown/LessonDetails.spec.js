import React from 'react';
import LessonDetails from '../../../../../src/container/courseBreakdown/subjectBreakdown/LessonBreakdown/LessonDetails';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from "history";
import configureMockStore from "redux-mock-store";
import {Main} from "../../../../../src/container/Main/Main";
import {Provider as ReduxProvider} from "react-redux";


const lesson = {
    id: 1,
    name: "test lesson name",
    lessonNumber: 1,
    pages: []
}

const course =  {
    name: "testCourseName",
    summary: "testCourseSummary",
    subjects: [
        {
            name: "testSubjectName",
            summary: "testSubjectSummary",
            subjectNumber: 1,
            lessons: [
                lesson
            ],
        }
    ]
}

const mockStore = configureMockStore()
const mockState = {
    lesson: lesson
}
let store = mockStore(mockState)

it('renders correctly', () => {
    const lesson = {
        name: "Threading the Machine",
        lessonNumber: 1,
    }

    const history = createMemoryHistory();
    history.location.state = {
        lesson: lesson
    }

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}><LessonDetails lesson={lesson}/></Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});