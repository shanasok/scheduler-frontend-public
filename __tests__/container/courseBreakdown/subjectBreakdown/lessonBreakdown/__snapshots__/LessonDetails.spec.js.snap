// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="lesson"
>
  <div>
    Threading the Machine
  </div>
  <a
    href="/lesson"
    onClick={[Function]}
  >
    <button
      className="lessonViewButton ribbonSlideButton curvedButton"
      onClick={[Function]}
    >
      Slides
    </button>
  </a>
</div>
`;
